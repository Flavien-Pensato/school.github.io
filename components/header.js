import React, { Component } from 'react';
import moment from 'moment';
import { Nav, Container } from 'react-bootstrap';
import styled from '@emotion/styled';
import Link from 'next/link';

import firebase from '../config/firebase';

import { DisplayContext } from '../modules/display/display.context';

moment.locale('fr');

const NavStyled = styled(Nav)`
  border-top: solid 1px black;
  border-bottom: solid 1px black;
`;

const HeaderTitle = styled.h1`
  font-size: 2rem;
  padding: 1rem 2rem;
  font-weight: 700;
  text-align: center;
  margin: 0;
  @media (max-width: 700px) {
    font-size: 1rem;
  }
`;

const LogoutButton = styled.button`
  font-size: 0.875rem;
  font-weight: 700;
  @media (max-width: 700px) {
    position: inherit;
    span {
      display: none;
    }
  }
`;

const LogoutSvg = () => (
  <StyledLogoutSvg>
    <path d="M0 0h24v24H0z" fill="none" />
    <path d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z" />
  </StyledLogoutSvg>
);

const StyledLogoutSvg = styled.svg`
  width: 24px;
  height: 24px;

  display: none;

  @media (max-width: 700px) {
    display: inherit;
  }
`;

const HeaderWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  @media (max-width: 700px) {
    justify-content: flex-end;
  }
`;

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = { menu: false };
  }

  handleClickMenu = event => {
    event.preventDefault();

    this.setState({ menu: !this.state.menu });
  };

  handleSignout = () => {
    firebase.auth().signOut();
  };

  render() {
    return (
      <Container>
        <HeaderWrapper>
          <HeaderTitle>Planning de la MFR de chatte</HeaderTitle>
          <LogoutButton onClick={this.handleSignout}>
            <span>Se d√©connecter</span>
            <LogoutSvg />
          </LogoutButton>
        </HeaderWrapper>
        <NavStyled>
          <Link href="/">
            <Nav.Link href="/">Accueil</Nav.Link>
          </Link>
          <Link href="/classes">
            <Nav.Link href="/classes">Classes</Nav.Link>
          </Link>
          <Link href="/calendrier">
            <Nav.Link href="/calendrier">Calendrier</Nav.Link>
          </Link>
          <Link href="/taches">
            <Nav.Link href="/taches">Taches</Nav.Link>
          </Link>
          <Link href="/">
            <Nav.Link href="/">
              <DisplayContext.Consumer>
                {({ date }) => (
                  <span>
                    Semaine du&nbsp;
                    {date.format('LL')}
                  </span>
                )}
              </DisplayContext.Consumer>
            </Nav.Link>
          </Link>
        </NavStyled>
      </Container>
    );
  }
}

export default Header;
