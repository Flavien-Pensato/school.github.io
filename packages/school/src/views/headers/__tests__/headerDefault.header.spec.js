import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { HeaderDefault } from '../headerDefault.header';

moment.locale('fr');

describe('Header default', () => {
  const props = {};
  const getWrapper = () => shallow(<HeaderDefault {...props} />);

  beforeEach(() => {
    Date.now = jest.fn(() => 1487076708000);
    props.currentWeek = moment();
    props.signOutAction = jest.fn();
    props.match = { params: {} };
  });

  it('should render correctly', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
