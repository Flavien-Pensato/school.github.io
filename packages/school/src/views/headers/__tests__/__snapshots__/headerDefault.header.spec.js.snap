// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header default should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderDefault
    currentWeek={"2017-02-14T12:51:48.000Z"}
    match={
      Object {
        "params": Object {},
      }
    }
    signOutAction={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.button
            onClick={[Function]}
          >
            <MenuSvg />
          </styled.button>
          <HeaderTitle>
            Planning de la MFR de chatte
          </HeaderTitle>
          <LogoutButton
            onClick={[MockFunction]}
          >
            <span>
              Se déconnecter
            </span>
            <LogoutSvg />
          </LogoutButton>
        </styled.div>,
        <styled.nav
          menu={false}
        >
          <styled.button
            onClick={[Function]}
          >
            <MenuClose />
          </styled.button>
          <HeaderLink
            to="/home"
          >
            Accueil
          </HeaderLink>
          <HeaderLink
            to="/classes"
          >
            Classes
          </HeaderLink>
          <HeaderLink
            to="/calendrier"
          >
            Calendrier
          </HeaderLink>
          <HeaderLink
            to="/taches"
          >
            Taches
          </HeaderLink>
          <HeaderLink
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/home"
          >
            Semaine du 
            14 février 2017
          </HeaderLink>
        </styled.nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.button
              onClick={[Function]}
            >
              <MenuSvg />
            </styled.button>,
            <HeaderTitle>
              Planning de la MFR de chatte
            </HeaderTitle>,
            <LogoutButton
              onClick={[MockFunction]}
            >
              <span>
                Se déconnecter
              </span>
              <LogoutSvg />
            </LogoutButton>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MenuSvg />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Planning de la MFR de chatte",
            },
            "ref": null,
            "rendered": "Planning de la MFR de chatte",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span>
                  Se déconnecter
                </span>,
                <LogoutSvg />,
              ],
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Se déconnecter",
                },
                "ref": null,
                "rendered": "Se déconnecter",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.button
              onClick={[Function]}
            >
              <MenuClose />
            </styled.button>,
            <HeaderLink
              to="/home"
            >
              Accueil
            </HeaderLink>,
            <HeaderLink
              to="/classes"
            >
              Classes
            </HeaderLink>,
            <HeaderLink
              to="/calendrier"
            >
              Calendrier
            </HeaderLink>,
            <HeaderLink
              to="/taches"
            >
              Taches
            </HeaderLink>,
            <HeaderLink
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/home"
            >
              Semaine du 
              14 février 2017
            </HeaderLink>,
          ],
          "menu": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MenuClose />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Accueil",
              "to": "/home",
            },
            "ref": null,
            "rendered": "Accueil",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Classes",
              "to": "/classes",
            },
            "ref": null,
            "rendered": "Classes",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Calendrier",
              "to": "/calendrier",
            },
            "ref": null,
            "rendered": "Calendrier",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Taches",
              "to": "/taches",
            },
            "ref": null,
            "rendered": "Taches",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "Semaine du ",
                "14 février 2017",
              ],
              "style": Object {
                "textDecoration": "none",
              },
              "to": "/home",
            },
            "ref": null,
            "rendered": Array [
              "Semaine du ",
              "14 février 2017",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.button
              onClick={[Function]}
            >
              <MenuSvg />
            </styled.button>
            <HeaderTitle>
              Planning de la MFR de chatte
            </HeaderTitle>
            <LogoutButton
              onClick={[MockFunction]}
            >
              <span>
                Se déconnecter
              </span>
              <LogoutSvg />
            </LogoutButton>
          </styled.div>,
          <styled.nav
            menu={false}
          >
            <styled.button
              onClick={[Function]}
            >
              <MenuClose />
            </styled.button>
            <HeaderLink
              to="/home"
            >
              Accueil
            </HeaderLink>
            <HeaderLink
              to="/classes"
            >
              Classes
            </HeaderLink>
            <HeaderLink
              to="/calendrier"
            >
              Calendrier
            </HeaderLink>
            <HeaderLink
              to="/taches"
            >
              Taches
            </HeaderLink>
            <HeaderLink
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/home"
            >
              Semaine du 
              14 février 2017
            </HeaderLink>
          </styled.nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.button
                onClick={[Function]}
              >
                <MenuSvg />
              </styled.button>,
              <HeaderTitle>
                Planning de la MFR de chatte
              </HeaderTitle>,
              <LogoutButton
                onClick={[MockFunction]}
              >
                <span>
                  Se déconnecter
                </span>
                <LogoutSvg />
              </LogoutButton>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <MenuSvg />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Planning de la MFR de chatte",
              },
              "ref": null,
              "rendered": "Planning de la MFR de chatte",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <span>
                    Se déconnecter
                  </span>,
                  <LogoutSvg />,
                ],
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Se déconnecter",
                  },
                  "ref": null,
                  "rendered": "Se déconnecter",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.button
                onClick={[Function]}
              >
                <MenuClose />
              </styled.button>,
              <HeaderLink
                to="/home"
              >
                Accueil
              </HeaderLink>,
              <HeaderLink
                to="/classes"
              >
                Classes
              </HeaderLink>,
              <HeaderLink
                to="/calendrier"
              >
                Calendrier
              </HeaderLink>,
              <HeaderLink
                to="/taches"
              >
                Taches
              </HeaderLink>,
              <HeaderLink
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                to="/home"
              >
                Semaine du 
                14 février 2017
              </HeaderLink>,
            ],
            "menu": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <MenuClose />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Accueil",
                "to": "/home",
              },
              "ref": null,
              "rendered": "Accueil",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Classes",
                "to": "/classes",
              },
              "ref": null,
              "rendered": "Classes",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Calendrier",
                "to": "/calendrier",
              },
              "ref": null,
              "rendered": "Calendrier",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Taches",
                "to": "/taches",
              },
              "ref": null,
              "rendered": "Taches",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "Semaine du ",
                  "14 février 2017",
                ],
                "style": Object {
                  "textDecoration": "none",
                },
                "to": "/home",
              },
              "ref": null,
              "rendered": Array [
                "Semaine du ",
                "14 février 2017",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
